lib/mods_display/fields/imprint.rb

```diff
@@ -47,167 +54,114 @@ def date_values(element)
     end
 
     class DateValue
-      def initialize(element)
-        @element = element
-      end
+        attr_reader :value
+        delegate :text, :date, :point, :qualifier, :encoding, to: :value
 
-      # True if the element text isn't blank or the placeholder "9999".
-      def valid?
-        @element.text.present? && @element.text.strip != '9999'
-      end
-
-      # Value of element's "point" attribute; nil if empty or not present.
-      # Points are "start" or "end" (if part of a date range)
-      def point
-        @element.attributes['point']&.value.presence
-      end
-
-      # Value of element's "qualifier" attribute; nil if empty or not present.
-      # Qualifiers are "approximate", "questionable", or "inferred"
-      def qualifier
-        @element.attributes['qualifier']&.value&.downcase.presence
-      end
-
-      # Value of element's "encoding" attribute; nil if empty or not present.
-      # Encodings are "w3cdtf", "iso8601", or "edtf"
-      def encoding
-        @element.attributes['encoding']&.value&.downcase.presence
-      end
-
-      # Element text reduced to digits and hyphen. Captures date ranges and
-      # negative (B.C.) dates. Used for comparison/deduping.
-      def base_value
-        @element.text.scan(/[\d-]/).join
-      end
-
-      # Decoded version of the date, if it was encoded. Strips leading zeroes.
-      def decoded_value
-        # Delegate to the appropriate decoding method, if any
-        decoded_date = case encoding
-                       when 'w3cdtf'
-                         decode_w3cdtf
-                       when 'iso8601'
-                         decode_iso8601
-                       when 'edtf'
-                         decode_edtf
-                       else
-                         @element.text.strip
-                       end
-
-        # Year zero is a special case; preserve it as "0"
-        return decoded_date if decoded_date == '0'
-
-        # Otherwise strip leading zeroes and return
-        decoded_date.gsub(/^0*/, '')
-      end
-
-      # Decoded version of the date with "B.C." or "A.D." appended.
-      def bc_ad_value
-        date = decoded_value
-
-        # Negative edtf dates are B.C. and may still have leading zeroes to remove
-        if encoding == 'edtf' && date.to_i < 1
-          year = date.gsub(/^-0*/, '').to_i + 1
-          "#{year} B.C."
-
-        # Any dates before the year 1000 are explicitly marked A.D.
-        elsif date.match?(/^\d{1,3}$/)
-          "#{date} A.D."
-
-        # Leave the value alone otherwise; A.D. is implied
-        else
-          date
+        def initialize(value)
+          @value = value
         end
```

Comment:
does this thing still get passed a `Nokogiri::XmlElement` or whatever (that it then hands off to the mods gem)? I haven't seen the `:delegate` above so I'm trying to interpret what this refactor changed.

Reply:
Yeah, same element but with the MODs  gem decorations. delegate is just a fancy way to write e.g.

```
def encoding
  value.encoding
end
```